src/rgw/rgw_rados.cc

```diff
@@ -6411,14 +6412,20 @@ int RGWRados::fetch_remote_obj(RGWObjectCtx& obj_ctx,
 
   string obj_name = dest_obj.bucket.name + "/" + dest_obj.get_object();
 
-  RGWOpStateSingleOp opstate(this, client_id, op_id, obj_name);
+  RGWOpStateSingleOp *opstate = NULL;
```

Comment:
it would be nice if we could avoid the heap allocation here.. it looks like you could leave opstate on the stack and replace the `if (opstate) {` blocks with `if (record_op_state) {`

otherwise, consider using std::unique_ptr to make sure we don't miss a delete in an error path